src/main/kotlin/org/opensearch/commons/alerting/model/Monitor.kt

```diff
@@ -0,0 +1,316 @@
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.alerting.settings.SupportedClusterMetricsSettings
+import org.opensearch.commons.alerting.util.IndexUtils.Companion.MONITOR_MAX_INPUTS
+import org.opensearch.commons.alerting.util.IndexUtils.Companion.MONITOR_MAX_TRIGGERS
+import org.opensearch.commons.alerting.util.IndexUtils.Companion.NO_SCHEMA_VERSION
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._ID
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._VERSION
+import org.opensearch.commons.alerting.util.instant
+import org.opensearch.commons.alerting.util.isBucketLevelMonitor
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.alerting.util.optionalUserField
+import org.opensearch.commons.authuser.User
+import java.io.IOException
+import java.time.Instant
+import java.util.Locale
+
+data class Monitor(
+    override val id: String = NO_ID,
+    override val version: Long = NO_VERSION,
+    override val name: String,
+    override val enabled: Boolean,
+    override val schedule: Schedule,
+    override val lastUpdateTime: Instant,
+    override val enabledTime: Instant?,
+    // TODO: Check how this behaves during rolling upgrade/multi-version cluster
+    //  Can read/write and parsing break if it's done from an old -> new version of the plugin?
+    val monitorType: MonitorType,
+    val user: User?,
+    val schemaVersion: Int = NO_SCHEMA_VERSION,
+    val inputs: List<Input>,
+    val triggers: List<Trigger>,
+    val uiMetadata: Map<String, Any>
+) : ScheduledJob {
+
+    override val type = MONITOR_TYPE
+
+    init {
+        // Ensure that trigger ids are unique within a monitor
+        val triggerIds = mutableSetOf<String>()
+        triggers.forEach { trigger ->
+            require(triggerIds.add(trigger.id)) { "Duplicate trigger id: ${trigger.id}. Trigger ids must be unique." }
+            // Verify Trigger type based on Monitor type
+            when (monitorType) {
+                MonitorType.QUERY_LEVEL_MONITOR ->
+                    require(trigger is QueryLevelTrigger) { "Incompatible trigger [${trigger.id}] for monitor type [$monitorType]" }
+                MonitorType.BUCKET_LEVEL_MONITOR ->
+                    require(trigger is BucketLevelTrigger) { "Incompatible trigger [${trigger.id}] for monitor type [$monitorType]" }
+                MonitorType.CLUSTER_METRICS_MONITOR ->
+                    require(trigger is QueryLevelTrigger) { "Incompatible trigger [${trigger.id}] for monitor type [$monitorType]" }
+                MonitorType.DOC_LEVEL_MONITOR ->
+                    require(trigger is DocumentLevelTrigger) { "Incompatible trigger [${trigger.id}] for monitor type [$monitorType]" }
```

Comment:
Can we take an attempt and decouple the Monitor model to separate out the pojo and execution logic. That way we can have a simpler (leaner) model here, required for consumers, whole the core logic is still held by the Alerting. 
This will prevent unnecessary dependencies being pulled in such as ``BucketLevelTriggers`` and ``BucketSelectorExt``. 

I am open to doing this incrementally as long we can add ToDos and open a github issue to do this as a followup